openssl/src/hash.rs

```diff
@@ -89,9 +149,12 @@ use self::State::*;
 ///
 /// # Warning
 ///
-/// Don't actually use MD5 and SHA-1 hashes, they're not secure anymore.
+/// The MD5 and SHA-1 hash algorithms are no longer considered secure and should not be used in new
+/// software. Use of these algorithms should be restricted only to cases where compatibility with
+/// legacy systems is required.
 ///
-/// Don't ever hash passwords, use the functions in the `pkcs5` module or bcrypt/scrypt instead.
+/// The hash algorithms available here are not suited to password hashing. For such applications,
+/// prefer instead to use either the `pkcs5` module or, more simply, either bcrypt or scrypt.
```

Comment:
I don't know if I'd say that bcrypt or scrypt are more simple. scrypt is in the `pkcs5` module, and pbkdf2 (also in the `pkcs5` module) is a decent third option.